web-app/src/components/DipMap.js

```diff
@@ -745,6 +745,110 @@ export default class DipMap extends React.Component {
 					this.debugCount("updateMap/hidProvince");
 				}
 			}
+
+
+			//Here we check each non-SC and non-Sea territory. If all surrounding SCs are of the same power and none is "Neutral", colour them that power.
+			//Get all nodes, disqualify Sea and SC, and per node collect the edges in an array.
+			for (let prov in this.state.variant.Properties.Graph.Nodes) {
+				const node = this.state.variant.Properties.Graph.Nodes[prov];
+				if (!node.SC && node.Subs[""].Flags.Land) {
+					const borderprovs = [];
+					for (let edge in node.Subs[""].Edges) {
+						const edgenode = this.state.variant.Properties.Graph.Nodes[edge];
+						if (typeof edgenode !== "undefined") {
+							if (edgenode.SC) {
+								borderprovs.push(SCs[edgenode.Name]);
+							}
+						}
+					}
+
+					//check if all members of array are equal/not neutral.
+					let shoulddraw = true;
+					for (let i = 0; i < borderprovs.length; i++) {
+						if (
+							borderprovs[i] !== borderprovs[0] ||
+							borderprovs[i] == "Neutral"
+						) {
+							shoulddraw = false;
+							break;
+						}
+					}
+
+					let countNeutral = 0;
+					for (let i = 0; i < borderprovs.length; i++) {
+						if (
+							typeof borderprovs[i] == "undefined"
+						) {
+						countNeutral ++;
+						}
+					}
+					if (countNeutral == borderprovs.length) {
+						shoulddraw = false;
+					}
+
+
+					//if all equal, draw the colour.
+					if (shoulddraw) {
+						const col = helpers.natCol(borderprovs[0], this.state.variant);
+						this.map.colorProvince(prov, col);
+					}
+
+
+					//if by the default rule we don't draw it, check if "special" rules apply.
+					else {
+					
+						//Need to escape if starting provinces have not been defined
```

Comment:
Create a variable here: `extraDominanceRules = this.props.variant.Properties.ExtraDominanceRules`